// temboo/Library/LinkedIn/PeopleAndConnections.js

var util = require("util");
var choreography = require("temboo/core/choreography");


/*
    GetMemberProfile
    Returns the standard default profile of the current user.
*/


var GetMemberProfile = function(session) {
    /*
        Create a new instance of the GetMemberProfile Choreo. A TembooSession object, containing a valid
        set of Temboo credentials, must be supplied.
	*/
    var location = "/Library/LinkedIn/PeopleAndConnections/GetMemberProfile"
    GetMemberProfile.super_.call(this, session, location);

    /*
    Define a callback that will be used to appropriately format the results of this Choreo.
    */
    var newResultSet = function(resultStream) {
        return new GetMemberProfileResultSet(resultStream);
    }

	/*
	Obtain a new InputSet object, used to specify the input values for an execution of this Choreo.
	*/
    this.newInputSet = function() {
        return new GetMemberProfileInputSet();
    }
    
	/*
	Execute this Choreo with the specified inputs, calling the specified callback upon success,
	and the specified errorCallback upon error.
	*/
    this.execute = function(inputs, callback, errorCallback) {
        this._execute(inputs, newResultSet, callback, errorCallback);
    }
}


/*
An InputSet with methods appropriate for specifying the inputs to the GetMemberProfile
Choreo. The InputSet object is used to specify input parameters when executing this Choreo.
*/

var GetMemberProfileInputSet = function() {
    GetMemberProfileInputSet.super_.call(this);
        /*
        Set the value of the APIKey input for this Choreo. ((required, string) The API Key provided by LinkedIn (AKA the Client ID).)
        */
        this.set_APIKey = function(value) {
            this.setInput("APIKey", value);
        }

        /*
        Set the value of the AccessTokenSecret input for this Choreo. ((required, string) The Access Token Secret retrieved during the OAuth process (AKA the OAuth User Secret).)
        */
        this.set_AccessTokenSecret = function(value) {
            this.setInput("AccessTokenSecret", value);
        }

        /*
        Set the value of the AccessToken input for this Choreo. ((required, string) The Access Token retrieved during the OAuth process (AKA the OAuth User Token).)
        */
        this.set_AccessToken = function(value) {
            this.setInput("AccessToken", value);
        }

        /*
        Set the value of the FieldSelector input for this Choreo. ((optional, string) A comma separated list of profile properties to return.)
        */
        this.set_FieldSelector = function(value) {
            this.setInput("FieldSelector", value);
        }

        /*
        Set the value of the ResponseFormat input for this Choreo. ((optional, string) The format that the response should be in. Valid values are: xml (the default) and json.)
        */
        this.set_ResponseFormat = function(value) {
            this.setInput("ResponseFormat", value);
        }

        /*
        Set the value of the SecretKey input for this Choreo. ((required, string) The Secret Key provided by LinkedIn (AKA the Client Secret).)
        */
        this.set_SecretKey = function(value) {
            this.setInput("SecretKey", value);
        }

}

/*
A ResultSet with methods tailored to the values returned by the GetMemberProfile Choreo.
The ResultSet object is used to retrieve the results of a Choreo execution.
*/

var GetMemberProfileResultSet = function(resultStream) {
    GetMemberProfileResultSet.super_.call(this, resultStream);    
        /*
        Retrieve the value for the "Response" output from this Choreo execution. (The response from LinkedIn.)
        */
        this.get_Response = function() {
            return this.getResult("Response");
        }
}

util.inherits(GetMemberProfile, choreography.Choreography);
util.inherits(GetMemberProfileInputSet, choreography.InputSet);
util.inherits(GetMemberProfileResultSet, choreography.ResultSet);
exports.GetMemberProfile = GetMemberProfile;
